client/layout/masterbar/masterbar-cart-button.tsx

```diff
@@ -0,0 +1,64 @@
+import { Popover } from '@automattic/components';
+import { CheckoutErrorBoundary } from '@automattic/composite-checkout';
+import { MiniCart } from '@automattic/mini-cart';
+import { useShoppingCart } from '@automattic/shopping-cart';
+import { useTranslate } from 'i18n-calypso';
+import { useRef, useState } from 'react';
+import MasterbarItem from './item';
+
+import './masterbar-cart-button-style.scss';
+
+export type MasterbarCartButtonProps = {
+	selectedSiteSlug: string | undefined;
+	goToCheckout: ( siteSlug: string ) => void;
+};
+
+export function MasterbarCartButton( {
+	selectedSiteSlug,
+	goToCheckout,
+}: MasterbarCartButtonProps ): JSX.Element | null {
+	const { responseCart, reloadFromServer } = useShoppingCart( selectedSiteSlug );
+	const masterbarButtonRef = useRef( null );
```

Comment:
Naming nit: does this make more sense to be called `masterbarCartButtonRef`? Or if we'd like to keep it simple, it can also be `cartButtonRef`?